internal/services/machinelearning/machine_learning_workspace_network_outbound_rule_private_endpoint_resource.go

```diff
@@ -0,0 +1,242 @@
+}
+
+func (r WorkspaceNetworkOutboundRulePrivateEndpoint) ModelObject() interface{} {
+	return &machineLearningWorkspaceOutboundRulePrivateEndpointModel{}
+}
+
+func (r WorkspaceNetworkOutboundRulePrivateEndpoint) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return managednetwork.ValidateOutboundRuleID
+}
+
+func (r WorkspaceNetworkOutboundRulePrivateEndpoint) Arguments() map[string]*pluginsdk.Schema {
+	arguments := map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"workspace_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: managednetwork.ValidateWorkspaceID,
+		},
+
+		"service_resource_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"sub_resource_target": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				"vault",
+				"amlworkspace",
+				"blob",
+				"table",
+				"queue",
+				"file",
+				"web",
+				"dfs",
+				"redisCache",
+			}, false),
+		},
+
+		"spark_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+			ForceNew: true,
+		},
+	}
+	return arguments
+}
+
+func (r WorkspaceNetworkOutboundRulePrivateEndpoint) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r WorkspaceNetworkOutboundRulePrivateEndpoint) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MachineLearning.ManagedNetwork
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model machineLearningWorkspaceOutboundRulePrivateEndpointModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			workspaceId, err := managednetwork.ParseWorkspaceID(model.WorkspaceId)
+			if err != nil {
+				return err
+			}
+			id := managednetwork.NewOutboundRuleID(subscriptionId, workspaceId.ResourceGroupName, workspaceId.WorkspaceName, model.Name)
+			existing, err := client.SettingsRuleGet(ctx, id)
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+				}
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return tf.ImportAsExistsError("azurerm_machine_learning_workspace_network_outbound_rule_private_endpoint", id.ID())
+			}
+
+			resId, err := resourceids.ParseAzureResourceID(model.ServiceResourceId)
+			if err != nil {
+				return err
+			}
+
+			supportType := false
+			if subTypes, ok := resourceTypeSupportSubResType[resId.Provider]; ok {
+				for _, typ := range subTypes {
```

Comment:
```suggestion
				for _, type := range subTypes {
```

Reply:
`type` is the in-built type of go, cannot name to type

Reply:
oops! right you are.